File: app/src/main/java/ru/p3tr0vich/fuel/BroadcastReceiverDatabaseChanged.java
package ru.p3tr0vich.fuel;

import android.content.Context;
import android.content.Intent;
import android.support.annotation.NonNull;
import android.support.v4.content.LocalBroadcastManager;

public abstract class BroadcastReceiverDatabaseChanged extends BroadcastReceiverLocalBase {

    private static final String ACTION = BuildConfig.APPLICATION_ID + ".ACTION_DATABASE_CHANGED";
    private static final String EXTRA_ID = BuildConfig.APPLICATION_ID + ".EXTRA_ID";

    @Override
    protected final String getAction() {
        return ACTION;
    }

    public static void send(@NonNull Context context, long id) {
        LocalBroadcastManager.getInstance(context).sendBroadcast(new Intent(ACTION)
                .putExtra(EXTRA_ID, id));
    }

    @Override
    public final void onReceive(Context context, Intent intent) {
        onReceive(intent.getLongExtra(EXTRA_ID, -1));
    }

    public abstract void onReceive(long id);
}

File: app/src/main/kotlin/ru/p3tr0vich/fuel/receivers/BroadcastReceiverDatabaseChanged.kt
package ru.p3tr0vich.fuel.receivers

import android.content.Context
import android.content.Intent
import androidx.localbroadcastmanager.content.LocalBroadcastManager
import ru.p3tr0vich.fuel.BuildConfig

abstract class BroadcastReceiverDatabaseChanged : BroadcastReceiverLocalBase() {
    override val action: String
        get() = ACTION

    override fun onReceive(context: Context, intent: Intent) {
        onReceive(intent.getLongExtra(EXTRA_ID, -1))
    }

    abstract fun onReceive(id: Long)

    companion object {
        private const val ACTION = "${BuildConfig.APPLICATION_ID}.ACTION_DATABASE_CHANGED"
        private const val EXTRA_ID = "${BuildConfig.APPLICATION_ID}.EXTRA_ID"

        @JvmStatic
        fun send(context: Context, id: Long) {
            send(context, Intent(ACTION).putExtra(EXTRA_ID, id))
        }
    }
}